concent_api/core/views.py

```diff
@@ -1220,6 +1225,632 @@ def store_message_and_message_status(
+            client_public_key,
+        )
+        file_transfer_token.operation = 'download'
+
+        response_to_client = message.concents.ForceGetTaskResultDownload(
+            file_transfer_token     = file_transfer_token,
+            force_get_task_result   = message.concents.ForceGetTaskResult(
+                report_computed_task = report_computed_task,
+            )
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.ForceSubtaskResults.name:  # pylint: disable=no-member
+        ack_report_computed_task = deserialize_message(database_message.subtask.ack_report_computed_task.data.tobytes())
+        response_to_client = message.concents.ForceSubtaskResults(
+            ack_report_computed_task = ack_report_computed_task
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.SubtaskResultsSettled.name:  # pylint: disable=no-member
+        task_to_compute = deserialize_message(database_message.subtask.task_to_compute.data.tobytes())
+        response_to_client = message.concents.SubtaskResultsSettled(
+            task_to_compute = task_to_compute,
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.ForceSubtaskResultsResponse.name:  # pylint: disable=no-member
+        subtask_results_accepted = deserialize_message(database_message.subtask.subtask_results_accepted.data.tobytes())
+        response_to_client = message.concents.ForceSubtaskResultsResponse(
+            subtask_results_accepted = subtask_results_accepted,
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.SubtaskResultsRejected.name:  # pylint: disable=no-member
+        subtask_results_rejected = deserialize_message(database_message.subtask.subtask_results_rejected.data.tobytes())
+        response_to_client = message.concents.ForceSubtaskResultsResponse(
+            subtask_results_rejected = subtask_results_rejected,
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.ForcePaymentCommitted.name:  # pylint: disable=no-member
+        payment_message = database_message.payments.filter(
+            pending_response__pk = database_message.pk
+        ).last()
+
+        response_to_client = message.concents.ForcePaymentCommitted(
+            payment_ts              = payment_message.payment_ts.timestamp(),
```

Comment:
Don't use `timestamp()` directly. Use the datetime conversion helper.

Reply:
done